java/compaction/compaction-core/src/test/java/sleeper/compaction/core/job/dispatch/CompactionJobDispatcherTest.java

```diff
@@ -270,6 +265,52 @@ void shouldFailForRetryWhenOneJobFailedToSend() {
                 compactionJobCreated(job1, Instant.parse("2024-11-15T10:30:10Z")));
         assertThat(delayedPendingQueue).isEmpty();
         assertThat(pendingDeadLetterQueue).isEmpty();
+        assertThat(s3PathToCompactionJobBatch).isNotEmpty();
+    }
+
+    @Test
+    void shouldDeleteAllFilesAfterBatchIsSent() {
+
+        // Given
+        FileReference file1 = fileFactory.rootFile("test1.parquet", 1234);
+        CompactionJob job1 = compactionFactory.createCompactionJob("test-job-1", List.of(file1), "root");
+        update(stateStore).addFiles(List.of(file1));
+        assignJobIds(List.of(job1));
+
+        CompactionJobDispatchRequest request = generateBatchRequestAtTime(
+                "test-batch", Instant.parse("2024-11-15T10:30:00Z"));
+        putCompactionJobBatch(request, List.of(job1));
+
+        // When
+        dispatchWithNoRetry(request);
+
+        // Then
+        assertThat(compactionQueue).containsExactly(job1);
+        assertThat(s3PathToCompactionJobBatch).isEmpty();
+    }
+
+    @Test
+    void shouldSendBatchToDeadLetterQueueWhenDeleteThrowsException() {
+
+        // Given
+        FileReference file1 = fileFactory.rootFile("test1.parquet", 1234);
+        CompactionJob job1 = compactionFactory.createCompactionJob("test-job-1", List.of(file1), "root");
+        update(stateStore).addFiles(List.of(file1));
+        assignJobIds(List.of(job1));
+
+        CompactionJobDispatchRequest request = generateBatchRequestAtTime(
+                "test-batch", Instant.parse("2024-11-15T10:30:00Z"));
+        putCompactionJobBatch(request, List.of(job1));
+
+        // When
+        deleteBatch = (bucketName, key) -> {
+            throw new RuntimeException();
+        };
+        dispatchWithNoRetry(request);
+
+        // Then
+        assertThat(pendingDeadLetterQueue).containsExactly(request);
+        assertThat(s3PathToCompactionJobBatch).isNotEmpty();
```

Comment:
It might be worth also asserting that the job was still sent to the compaction queue.